azurerm/resource_arm_dns_a_record.go

```diff
@@ -53,124 +53,121 @@ func resourceArmDnsARecord() *schema.Resource {
 	}
 }
 
-func resourceArmDnsARecordCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+func resourceArmDnsARecordCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	dnsClient := meta.(*ArmClient).dnsClient
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	zoneName := d.Get("zone_name").(string)
+	ttl := int64(d.Get("ttl").(int))
 
 	tags := d.Get("tags").(map[string]interface{})
-	expandedTags := expandTags(tags)
-
-	createCommand := &dns.CreateARecordSet{
-		Name:              d.Get("name").(string),
-		Location:          "global",
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ZoneName:          d.Get("zone_name").(string),
-		TTL:               d.Get("ttl").(int),
-		Tags:              *expandedTags,
-	}
+	metadata := expandTags(tags)
 
-	recordStrings := d.Get("records").(*schema.Set).List()
-	records := make([]dns.ARecord, len(recordStrings))
-	for i, v := range recordStrings {
-		records[i] = dns.ARecord{
-			IPv4Address: v.(string),
-		}
+	records, err := expandAzureRmDnsARecords(d)
+	props := dns.RecordSetProperties{
+		Metadata: metadata,
+		TTL:      &ttl,
+		ARecords: &records,
 	}
-	createCommand.ARecords = records
-
-	createRequest := rivieraClient.NewRequest()
-	createRequest.Command = createCommand
 
-	createResponse, err := createRequest.Execute()
-	if err != nil {
-		return fmt.Errorf("Error creating DNS A Record: %s", err)
-	}
-	if !createResponse.IsSuccessful() {
-		return fmt.Errorf("Error creating DNS A Record: %s", createResponse.Error)
+	parameters := dns.RecordSet{
+		Name:                &name,
+		RecordSetProperties: &props,
 	}
 
-	readRequest := rivieraClient.NewRequest()
-	readRequest.Command = &dns.GetARecordSet{
-		Name:              d.Get("name").(string),
-		ResourceGroupName: d.Get("resource_group_name").(string),
-		ZoneName:          d.Get("zone_name").(string),
-	}
-
-	readResponse, err := readRequest.Execute()
+	//last parameter is set to empty to allow updates to records after creation
+	// (per SDK, set it to '*' to prevent updates, all other values are ignored)
+	resp, err := dnsClient.CreateOrUpdate(resGroup, zoneName, name, dns.A, parameters, "", "")
```

Comment:
what's the penultimate argument for here? when there's more than one empty string we tend to separate these out into variables i.e.:
```
foo := "bar"
bar := "foo"
...CreateOrUpdate(..., foo, bar)
```